Client/Helpers/FxALoginHelper.swift

```diff
@@ -136,10 +136,14 @@ class FxALoginHelper {
     // It manages the asking for user permission for notification and registration 
     // for APNS and WebPush notifications.
     func application(_ application: UIApplication, didReceiveAccountJSON data: JSON) {
-        if data["keyFetchToken"].stringValue() == nil || data["unwrapBKey"].stringValue() == nil {
-            // The /settings endpoint sends a partial "login"; ignore it entirely.
+        if data["keyFetchToken"] == JSON.null || data["unwrapBKey"] == JSON.null {
+            // If a partial "login"(missings keys) is detected, this typically implies
+            // an unverified account. In this case, we want to defer control to the
+            // FxA content server to display the correct verification screen. Once account
+            // has been verified, a login message is sent, view transititions to /settings
+            // and syncing begins.
```

Comment:
Ok. I think I understand. 

However, I'm still not sure we want to do this just here:

 * if the account has been verified, can we/should we register for push notifications yet?
 * what happens when the user leaves the screen manually before verifying.

Reply:
> if the account has been verified, can we/should we register for push notifications yet?

When testing, the user gets presented with the notification permission modal as soon as they have a successful login (regardless if account has been confirmed or not). I thought there was some logic under the hood the hooked up the notifications and the account, despite account verification state.

> what happens when the user leaves the screen manually before verifying.

If they leave screen, they can go back by clicking the account indicator. Which would give them an option to resend verification emails etc.

Reply:
> I thought there was some logic under the hood the hooked up the notifications and the account, despite account verification state.

Yes, there is, but it happens in that method you've just prematurely returned from:

https://github.com/mozilla-mobile/firefox-ios/blob/master/Client/Helpers/FxALoginHelper.swift#L158

Reply:
@jhugman From our previous conversation, we wanted to move the logic that does this into  the `FxAContentViewController`. After digging more into it, it turns out that this was not really needed in either case. The FxA webview correctly sends the appropriate login messages and handles when the account is unverified.

For unverified accounts and logins, it shows the correct screen and setup push notifications if permission wasn't already granted.

Also, removing this logic fixes the current issue because it allows the user to navigate back and forth from FxiOS settings and `FxAContentViewController`.